### PR TITLE
NavBar2 (front-end)

### PR DESCRIPTION
**Pull request by Khal-Shah**
image

From what I gather, this needs further work involving user sessions/authentications for the texts and buttons to change when user is signed in.

When back end is fully implemented and there is a helper method that can be used to check if a user is signed in, the ask question button should turn green and redirect to new question page and the sign in text on NavBar should change to sign out.

I looked around but couldn't figure out how to get the back end done, but we need something like section 8.2.4 here:
http://rails-3-2.railstutorial.org/book/sign_in_out

Modified files: application.scss and application.html.erb (as navbar is supposed to be present on every page). Also put yield in container to center the content of page.

### CODE REVIEW

**Comment by samgo1** [comment]
@ThierrySt-Arnaud , I put you as reviewer because the navbar has merge conflicts with the user panel stuff, can you check that out and resolve conflicts? When everything looks good you can merge it to dev

File: QA-webapp/app/assets/stylesheets/application.scss [hunk 1]
```diff
@@ -6,3 +6,43 @@
 @import "users";
 @import "welcome";
 @import "userpanel";
+
```
Comment (ThierrySt-Arnaud) [inline QA-webapp/app/assets/stylesheets/application.scss]: Everything after this should be put into another file (say navbar.scss) and then an @import statement should be added here (if the file is navbar.scss, then it should be `@import "navbar";`). This way you can modify all the css for the navbar in the one place without potentially breaking the rest and changing the order for the css file is much easier.

**Review by ThierrySt-Arnaud — CHANGES_REQUESTED** [review]
I changed the order of a few lines, but otherwise everything should work as is. Just one thing I would change. Its nitpicking but I think it will save us a lot of trouble down the line; we can easily separate the css files so that we don't have all the css in one file that will constantly be conflicting as we modify it. I put instructions in a comment.

**Comment by Khal-Shah** [comment]
I put the styling in a different file as requested. 

Fixed the padding issue, changed background color. 

The Sign In button does change to Sign Out once an user is created, but I still couldn't find the delete session path so the sign out button currently gets routed to home page when user signs out.

The question button changes to green from red upon user creation and routes to new question page when user clicks on it.
image

What still needs to be fixed: once user clicks on the green question button, it changes back to red in the new question page. Same for the Sign Out button (changes back to Sign In).

Also I still get an InvalidAuthenticityToken error when trying to post a question.

I haven't added a "List Questions" item on the navbar, I can try to do that tomorrow given we can make what we currently have functional first. Otherwise, that seems more like a Sprint 3 thing.

**Review by samgo1 — CHANGES_REQUESTED** [review]
We don't need the additional database entity attributes for the question and we don't need a 'views' table in the db. Can you remove the addition from QA-webapp/db/schema.rb?

**Review by ThierrySt-Arnaud — APPROVED** [review]
Everything looks good for me.

**Comment by Khal-Shah** [comment]
I'm not sure how those got into there...But I removed everything new that were added to schema.rb

**Comment by Khal-Shah** [comment]
Added a static show question button that shows the listing of questions at /questions:
image

**Review by samgo1 — APPROVED** [review]

**Pull request merged**